### PR TITLE
Switch pdb to podgroup for gang scheduling

### PR DESCRIPTION
**Pull request by fisherxu**
Pdb has been deprecated in kube-batch.

 
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/mpi-operator/111)

### CODE REVIEW

File: pkg/controllers/v1alpha2/mpi_job_controller.go [hunk 1]
```diff
@@ -522,13 +525,13 @@ func (c *MPIJobController) getLauncherJob(mpiJob *kubeflow.MPIJob) (*batchv1.Job
 	return launcher, nil
 }
 
-// getOrCreatePDB will create a PDB for gang scheduling by kube-batch.
-func (c *MPIJobController) getOrCreatePDB(mpiJob *kubeflow.MPIJob, minAvailableWorkerReplicas int) (*policyv1beta1.PodDisruptionBudget, error) {
+// getOrCreatePodGroups will create a PodGroup for gang scheduling by kube-batch.
+func (c *MPIJobController) getOrCreatePodGroups(mpiJob *kubeflow.MPIJob, minAvailableWorkerReplicas int32) (*podgroupv1alpha2.PodGroup, error) {
```
Comment (fisherxu) [inline pkg/controllers/v1alpha2/mpi_job_controller.go]: Now the count of worker will not be set to 0 when the training work completed. So only has the CreatePodGroup func, will add DeletePodGroup alone with set worker to 0 in follow-up PRs

File: pkg/controllers/v1alpha2/mpi_job_controller.go [hunk 2]
```diff
@@ -598,7 +601,7 @@ func (c *MPIJobController) getOrCreateLauncherServiceAccount(mpiJob *kubeflow.MP
 }
 
 // getOrCreateLauncherRole gets the launcher Role controlled by this MPIJob.
-func (c *MPIJobController) getOrCreateLauncherRole(mpiJob *kubeflow.MPIJob, workerReplicas int) (*rbacv1.Role, error) {
+func (c *MPIJobController) getOrCreateLauncherRole(mpiJob *kubeflow.MPIJob, workerReplicas int32) (*rbacv1.Role, error) {
```
Comment (fisherxu) [inline pkg/controllers/v1alpha2/mpi_job_controller.go]: let's switch int to int32.

File: cmd/mpi-operator.v1alpha2/app/server.go [hunk 3]
```diff
@@ -42,6 +41,9 @@ import (
 	mpijobclientset "github.com/kubeflow/mpi-operator/pkg/client/clientset/versioned"
 	informers "github.com/kubeflow/mpi-operator/pkg/client/informers/externalversions"
 	"github.com/kubeflow/mpi-operator/pkg/version"
+	kubebatchclient "github.com/kubernetes-sigs/kube-batch/pkg/client/clientset/versioned"
```
Comment (gaocegege) [inline cmd/mpi-operator.v1alpha2/app/server.go]: Can we place the deps in the second group? I think the third-party packages are always in the second group.

File: pkg/controllers/v1alpha2/mpi_job_controller.go [hunk 4]
```diff
@@ -52,6 +48,10 @@ import (
 	kubeflowScheme "github.com/kubeflow/mpi-operator/pkg/client/clientset/versioned/scheme"
 	informers "github.com/kubeflow/mpi-operator/pkg/client/informers/externalversions/kubeflow/v1alpha2"
 	listers "github.com/kubeflow/mpi-operator/pkg/client/listers/kubeflow/v1alpha2"
+	podgroupv1alpha2 "github.com/kubernetes-sigs/kube-batch/pkg/apis/scheduling/v1alpha2"
+	kubebatchclient "github.com/kubernetes-sigs/kube-batch/pkg/client/clientset/versioned"
+	podgroupsinformer "github.com/kubernetes-sigs/kube-batch/pkg/client/informers/externalversions/scheduling/v1alpha2"
+	podgroupslists "github.com/kubernetes-sigs/kube-batch/pkg/client/listers/scheduling/v1alpha2"
```
Comment (gaocegege) [inline pkg/controllers/v1alpha2/mpi_job_controller.go]: Same here. The third group only places pakcages in this repository `github.com/kubeflow/mpi-operator`

**Pull request merged**